### PR TITLE
HPCC-23168 Roxie internal error on spill file that is never read

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Compiled.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review. I think this is (at least close to) the correct fix but have not ben able to test it as I don't know how to reproduce the issue.

**Comment by ghalliday** [comment]
@richardkchapman please can you include the test case I created.

**Comment by richardkchapman** [comment]
@ghalliday Done

File: testing/regress/ecl/issue23168.ecl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+import dbglog from Std.System.log;
+
+#option ('targetClusterType', 'roxie');
```
Comment (ghalliday) [inline testing/regress/ecl/issue23168.ecl]: This line needs to be removed

**Comment by ghalliday** [comment]
@richardkchapman the example needs a bit more work.  I had wondered about the reset without stop, but had convinced myself it should call stop.  Looks like I was wrong.

**Comment by richardkchapman** [comment]
@ghalliday Please review

**Comment by ghalliday** [comment]
@richardkchapman this looks plausible, but I would be wary of merging the change:

`      return dependentCount == 0;  // Codegen normally optimizes...
`
that far back - I wonder if it could break anything.  What do you think.

Also needs squashing.

**Comment by richardkchapman** [comment]
@ghalliday  It could potentially break something (but hopefully not). Depends on whether the original reporter is still hitting the issue or were they able to work around via a change to the ECL or codegen?

**Comment by ghalliday** [comment]
@richardkchapman I'm relatively happy to merge this.  Can you squash.

**Comment by richardkchapman** [comment]
@ghalliday Squashed

**Pull request merged**